modules/EvseV2G/iso_server.cpp

```diff
@@ -277,27 +211,28 @@ static void printMbedVerifyErrorCode(int AErr, uint32_t AFlags) {
 }
 
 /*!
- * \brief check_iso1_signature This function validates the ISO signature
- * \param iso1_signature is the signature of the ISO EXI fragment
+ * \brief check_iso2_signature This function validates the ISO signature
+ * \param iso2_signature is the signature of the ISO EXI fragment
  * \param public_key is the public key to validate the signature against the ISO EXI fragment
- * \param iso1_exi_fragment iso1_exi_fragment is the ISO EXI fragment
+ * \param iso2_exi_fragment iso2_exi_fragment is the ISO EXI fragment
  */
-static bool check_iso1_signature(const struct iso1SignatureType* iso1_signature, mbedtls_ecdsa_context* public_key,
-                                 struct iso1EXIFragment* iso1_exi_fragment) {
+static bool check_iso2_signature(const struct iso2_SignatureType* iso2_signature, mbedtls_ecdsa_context* public_key,
+                                 struct iso2_exiFragment* iso2_exi_fragment) {
     /** Digest check **/
     int err = 0;
-    const struct iso1SignatureType* sig = iso1_signature;
+    const struct iso2_SignatureType* sig = iso2_signature;
     unsigned char buf[MAX_EXI_SIZE];
-    size_t buffer_pos = 0;
-    const struct iso1ReferenceType* req_ref = &sig->SignedInfo.Reference.array[0];
-    bitstream_t stream = {MAX_EXI_SIZE, buf, &buffer_pos, 0, 8 /* Set to 8 for send and 0 for recv */};
+    const struct iso2_ReferenceType* req_ref = &sig->SignedInfo.Reference.array[0];
+    exi_bitstream_t stream;
+    exi_bitstream_init(&stream, buf, MAX_EXI_SIZE, 8, NULL);
```

Comment:
Please change the 8 to a zero. That was the reason why PnC didn't work. The Exi fragment was encoded with an 8 byte offset, so of course the digest value was no longer correct.

Reply:
Well. That's embarressing.
I didn't even copy paste that. I actually typed 8 for some reason.